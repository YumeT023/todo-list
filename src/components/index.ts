export * from "./container";
export * from "./item";
export * from "./text-input";
export * from "./list-item";
