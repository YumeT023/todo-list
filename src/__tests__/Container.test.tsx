import { create } from "react-test-renderer";
import { Container } from "../components";

describe("Container", () => {
  describe("render", () => {
    test("rendered correctly", () => {
      const container = create(<Container />);
      expect(container).toBeTruthy();
    });

    test("[snapshot] rendered correctly", () => {
      const container = create(<Container />);
      const toJson = container.toJSON();
      expect(toJson).toMatchSnapshot();
    });
  });

  describe("render with props", () => {
    const margin = "4rem";
    const padding = "6rem";

    test("rendered with the given props", () => {
      const app = create(<Container m={margin} p={padding} />);
      const main = app.root;

      expect(main.props.p).toEqual(padding);
      expect(main.props.m).toEqual(margin);
    });

    test("[snapshot] rendered with the given props", () => {
      const toJson = create(<Container m={margin} p={padding} />).toJSON();
      expect(toJson).toMatchSnapshot();
    });
  });

  describe("render with additional style", () => {
    const containerStyle = {
      backgroundColor: "black",
      color: "white",
    };

    test("renders with the given props", () => {
      const app = create(<Container style={containerStyle} />);
      const container = app.root;

      const actualStyle = container.props.style;
      expect(actualStyle).toEqual(containerStyle);
    });

    test("[snapshot] renders with the given props", () => {
      const app = create(<Container style={containerStyle} />);

      const json = app.toJSON();

      expect(json).toMatchSnapshot();
    });
  });
});
