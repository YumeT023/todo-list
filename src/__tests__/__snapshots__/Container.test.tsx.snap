// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container render [snapshot] rendered correctly 1`] = `
<div
  style={
    Object {
      "margin": undefined,
      "padding": undefined,
    }
  }
/>
`;

exports[`Container render with additional style [snapshot] renders with the given props 1`] = `
<div
  style={
    Object {
      "backgroundColor": "black",
      "color": "white",
      "margin": undefined,
      "padding": undefined,
    }
  }
/>
`;

exports[`Container render with props [snapshot] rendered with the given props 1`] = `
<div
  style={
    Object {
      "margin": "4rem",
      "padding": "6rem",
    }
  }
/>
`;
